runtime/doc/api.txt

```diff
@@ -3108,6 +3108,199 @@ nvim_tabpage_set_var({tabpage}, {name}, {value})
+                              |autocmd-groups|.
+                            • event (string|array) required: event or events
+                              to match against |autocmd-events|.
+                            • pattern (string|array) optional: pattern or
+                              patterns to match against |autocmd-pattern|.
+                            • buffer (integer) optional: buffer number for
+                              buffer local autocmds (can't be used together
+                              with pattern) |autocmd-buflocal|.
+                            • desc (string) optional: description of the
+                              autocmd.
+                            • callback (function|string) optional: Lua
+                              function or Vim function (as string) to execute
+                              on event.
+                            • command (string) optional: Vim command to
+                              execute on event (can't be used together with
+                              callback).
+                            • once (boolean) optional: defaults to false. Run
+                              the autocmd only once |autocmd-once|.
+                            • nested (boolean) optional: defaults to false.
+                              Run nested autocmds |autocmd-nested|.
+
+                Return: ~
+                    Integer id of the created autocommand.
+
+                See also: ~
+                    |autocmd|
+                    |nvim_del_autocmd()|
+
+nvim_del_augroup_by_id({id})                        *nvim_del_augroup_by_id()*
+                Delete an autocommand group by id.
+
+                NOTE: behavior differs from |augroup-delete|. When deleting an
+                augroup, autocmds contained by this augroup will also be
+                deleted and cleared. This augroup will no longer exist
+
+                Parameters: ~
+                    {id}  Integer The id returned by nvim_create_augroup.
+
+                See also: ~
+                    |nvim_del_augroup_by_name()|
+                    |nvim_create_augroup()|
+
+nvim_del_augroup_by_name({name})                  *nvim_del_augroup_by_name()*
+                Delete an autocommand group by name.
+
+                NOTE: behavior differs from |augroup-delete|. When deleting an
+                augroup, autocmds contained by this augroup will also be
+                deleted and cleared. This augroup will no longer exist.
+
+                See also: ~
+                    |autocommand-groups|
+
+nvim_del_autocmd({id})                                    *nvim_del_autocmd()*
+                Delete an autocommand by id.
+
+                Parameters: ~
+                    {id}  Integer The id returned by nvim_create_autocmd
+
+                See also: ~
+                    |nvim_create_autocmd()|
+
+nvim_do_autocmd({*opts})                                   *nvim_do_autocmd()*
```

Comment:
Why did we introduce a new API verb "do" instead of using an existing verb like "exec"? Leaking the internal API just perpetuates the awkwardness of the internal API forever.

Reply:
We can still change it before release, I also found exec reading better that's why I describe it like that.